### PR TITLE
reimagining the NewChallengeScreen to improve usability and streamline challenge management

### PR DESCRIPTION
**Pull request by samymessouak**
## Description

This pull request focuses on reimagining the `NewChallengeScreen` to improve usability and streamline challenge management:

- **Removed "Friends Challenge" section**  
  - Paving the way for a potential redesign that may reintroduce this feature in the future.
- **Introduced "Your Own Challenge" section**  
  - Prioritizing individual challenge management for better usability.
- **Designed and implemented a scrollable box for ongoing challenges**  
  - Enhancing user experience by making challenges easier to locate and manage.
- **Optimized challenge handling process**  
  - Ensuring alignment with the updated UI/UX requirements.
- **Conducted thorough testing**  
  - Ensured seamless functionality and proper data synchronization with Firestore.

*This redesign lays the groundwork for more advanced challenge features while maintaining scalability for future updates.*

***NO WORRIES, THE TESTS ARE COMING IN THE NEXT PR, THIS IS FOR COACH BEING ABLE TO SEE ADVANCEMENT, FEEL FREE TO TRY IT IN THE BRANCH***

### CODE REVIEW

**Review by nada-ezziti — APPROVED** [review]
_**General Overview**_

The redesigned NewChallengeScreen brings valuable improvements in usability, organization, and functionality, especially for managing individual challenges. The code structure appears well-organized and aligns with the current UI/UX requirements. The integration of a scrollable box for ongoing challenges is a user-friendly addition, making it easy for users to track and manage their active challenges. 

___ 

**_Detailed Feedback_**

**1. Removal of the "Friends Challenge" Section**

- Removing this section is logical for now, given that the focus is on individual challenges. The flexibility to reintroduce it later allows for iterative development.
- Consider documenting any plans or requirements for reintroducing "Friends Challenge" in future sprints, as this will help align team expectations and guide future development.


**2. Introduction of "Your Own Challenge" Section**

- The addition of "Your Own Challenge" is clear and intuitive. To further improve user engagement, consider adding a brief intro message or icon in this section, helping new users understand its purpose immediately.


**3. Scrollable Ongoing Challenges Box**

- Implementing a scrollable box makes ongoing challenges easily accessible, even as the list grows.
-  It might be beneficial to add a subtle indication (like a fade effect at the bottom of the box) to show users they can scroll. 


**4. Optimized Challenge Handling**

- It’s great to see optimization efforts in handling challenges. Code related to user and challenge data in Firestore is well-organized.


**5. Testing and Future Test Plans** 
For upcoming tests, it would be helpful to focus on:
- Edge cases, such as handling large numbers of ongoing challenges.
- Data synchronization checks, ensuring any updates in Firestore reflect instantly in the UI.
- User experience under intermittent network conditions, especially for Firestore-dependent features.

- It would be better to push the tests with the code to ensure a safe merging.

____

_**Overall Impression**_

Great work on iteratively improving the screen’s functionality. I think it can be merged since you said that the tests are coming in the next PR.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt [hunk 1]
```diff
@@ -75,41 +99,113 @@ fun NewChallengeScreen(navigationActions: NavigationActions) {
...
+                    color = colorResource(R.color.dark_gray))
+              }
+          IconButton(
+              onClick = onDeleteClick,
+              modifier =
+                  Modifier.padding(start = 8.dp).testTag("DeleteButton${challenge.challengeId}")) {
+                Icon(
+                    imageVector = Icons.Default.Delete,
+                    contentDescription = "Delete Challenge",
+                    tint = Color.Red)
```
Comment (nada-ezziti) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt:207]: I think it's better to change the color of this icon to dark_gray to be more consistent with the Figma design.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt [hunk 2]
```diff
@@ -51,17 +41,24 @@ fun CreateAChallengeScreen(
...
               // Title
-              Text(text = "Select a Friend to Challenge", fontSize = 24.sp)
+              Text(
+                  text = "Select a Friend to Challenge",
+                  fontSize = 24.sp,
+                  modifier = Modifier.testTag("ChallengeTitle"))
 
               Spacer(modifier = Modifier.height(16.dp))
 
               // List of Friends
```
Comment (nada-ezziti) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt:59]: The friend list integration looks good. As a UX improvement, you could display a 'No friends available' message when the list is empty to guide the user.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt [hunk 3]
```diff
@@ -71,7 +68,8 @@ fun CreateAChallengeScreen(
                               selectedFriendId = friendId
                               showDialog = true
                             },
-                            modifier = Modifier.padding(8.dp)) {
+                            modifier =
+                                Modifier.padding(8.dp).testTag("ChallengeButton_$friendId")) {
                               Text("Challenge")
                             }
                       }
```
Comment (nada-ezziti) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt:75]: The dialog design is clean and functional. It might be helpful to add visual feedback (e.g., a highlight or animation) when a mode is selected to improve clarity.

**Review by ziyadbagh — APPROVED** [review]
Since there are no tests yet, I tested the challenge screens locally on your branch. I was able to send and receive challenges, and the scrollable box functions as expected.
